src/pages/index.js

```diff
@@ -121,22 +140,98 @@ function handleOverlay(evt) {
 
 function handleEditProfileSubmit(evt) {
   evt.preventDefault();
-  profileNameEl.textContent = nameInput.value;
-  profileDescriptionEl.textContent = descriptionInput.value;
-  closeModal(editModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({ name: nameInput.value, about: descriptionInput.value })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = {
-    name: cardNameInput.value,
-    link: cardLinkInput.value,
-  };
-  evt.target.reset();
-  const cardEl = getCardElement(inputValues);
-  cardList.prepend(cardEl);
-  disableButton(cardSubmitBtn, config);
-  closeModal(cardModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .appendCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((data) => {
+      const cardEl = getCardElement(data);
+      cardList.prepend(cardEl);
+      evt.target.reset();
+      disableButton(cardSubmitBtn, config);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileImage.src = data.avatar;
+      evt.target.reset();
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleLike(evt, id) {
+  let isLiked = evt.target.classList.contains("card__like-button_liked");
```

Comment:
`let isLiked`

If a variable is declared using `let`, it must be **reassigned** somewhere.
If a variable doesn’t change (is not reassigned), it is declared using `const`, to be 100% sure in its value.

**All your variables** should be declared using `const`.